### ISSUE TITLE
[Feature request] Control usernames in teamSpeak

### ISSUE THREAD

**Issue by tigranm75** [warlof/seat-connector#40]
Hi!

Please add nickname control mechanism to TS plugin. It can check by cron username in TS and kick users who don't meet username format requirements like discord "[%1$s] %2$s" with message "Your nickname must be [%1$s] %2$s"

Thank you!

**Comment by Nfinished** [comment]
As far as I know, Alliance Auth manages to exert direct control over Teamspeak users' names. Is that something SeAT could achieve? Alternatively, it doesn't seem to be _too_ much of a security issue to have users join programmatically with their correct name rather than a uid.

Caveat: Prefixes may not be as desirable for Teamspeak as they are for discord/discourse, since they can instead be set via Teamspeak group.

**Comment by warlof** [comment]
Teamspeak API does not allow client user to be altered, in any way.
Also, adding a specific rule for a driver would impact other driver, which is not expected.

We may implement a kick action on the update nickname flow for teamspeak, that's a working use case. But it seems a little bit hard for that poor boy, no ?

**Comment by tigranm75** [comment]
> We may implement a kick action on the update nickname flow for teamspeak, that's a working use case. But it seems a little bit hard for that poor boy, no ?

I think it's normal. Most of big alliances use that algorithm and people are familiar with that mechanic.

**Comment by tribulex** [comment]
It would be great to have the option to assign the user the correct name on initial login rather than a random series of characters. This way the onus is not on the user to change their name to a perfect format.

**Comment by Nfinished** [comment]
Yeah, it's not as "secure" but the risk seems incredibly minimal.

**Comment by tribulex** [comment]
If it is an option, accepting the risk is up to the person running the service.

**Comment by akturis** [comment]
ServerQuery can update the client description (clientdbedit cldbid=$ client_description=$).
The plugin can save the username with tag there.

**Comment by teh-banana** [comment]
Could we have an option to have SeAT continue to authenticate how it currently does (random letters / numbers), and then when running `seat-connector:apply:policies` drop server groups if the users nickname does not match the specified format. Having an option specific for TeamSpeak for this to be on/off would be ideal for people who want to continue using it as is.

**Comment by warlof** [comment]
@teh-banana I had spawn that at some time - but never be able to valid its flow : https://github.com/warlof/seat-teamspeak/pull/25
